stan/math/opencl/kernel_generator/scalar.hpp

```diff
@@ -88,22 +91,12 @@ class scalar_ : public operation_cl<scalar_<T>, T> {
   inline int cols() const { return base::dynamic; }
 
   /**
-   * View of a matrix that would be the result of evaluating this expression.
-   * @return view
-   */
-  inline matrix_cl_view view() const { return matrix_cl_view::Entire; }
-
-  /**
-   * Determine index of bottom diagonal written.
-   * @return number of columns
+   * Determine indices of extreme sub- and superdiagonals written.
+   * @return pair of indices - bottom and top diagonal
    */
-  inline int bottom_diagonal() const { return std::numeric_limits<int>::min(); }
-
-  /**
-   * Determine index of top diagonal written.
-   * @return number of columns
-   */
-  inline int top_diagonal() const { return std::numeric_limits<int>::max(); }
+  inline std::pair<int, int> extreme_diagonals() const {
+    return {std::numeric_limits<int>::min(), std::numeric_limits<int>::max()};
```

Comment:
I'm not really following the code here. If this is supposed to grab the extreme diagonal indices why is it returning the min and max of int types?

Reply:
The scalar expression does not have size. In a binary expression it is always broadcast in both dimensions to match the other argument. So we don't know what exactely its extreme diagonals are. By returning min and max  int we are saying that regardless of size it is broadcast to it covers whole matrix - there are no zero parts. Broadcasting also returns min or max to similar effect.